applications/crossbar/src/modules/cb_notifications.erl

```diff
@@ -255,6 +262,38 @@ disallow_delete(Context, Id) ->
         ],
     cb_context:add_validation_error(Id, <<"disallow">>, wh_json:from_list(Resp), Context).
 
+-spec may_be_validate_recipient_id(cb_context:context()) -> cb_context:context().
+may_be_validate_recipient_id(Context) ->
+    case cb_context:req_value(Context, <<"recipient_id">>) of
+        'undefined' ->
+            cb_context:set_resp_status(Context, 'success');
+        <<RecipientId:32/binary>> ->
+            validate_recipient_id(RecipientId, Context);
+        _ ->
+            Context
+    end.
+
+-spec validate_recipient_id(ne_binary(), cb_context:context()) -> cb_context:context().
+validate_recipient_id(RecipientId, Context) ->
```

Comment:
I think you can use `wh_util:is_in_account_hierarchy(SenderId, RecipientId)` here. It check's `RecipientId`'s account tree (`pvt_tree`) for SenderId.